youtube_dl/extractor/tvp.py

```diff
@@ -1,139 +1,247 @@
-        if series_title:
-            title = '%s, %s' % (series_title, title)
-
-        thumbnail = self._search_regex(
-            r"poster\s*:\s*'([^']+)'", webpage, 'thumbnail', default=None)
-
-        video_url = self._search_regex(
-            r'0:{src:([\'"])(?P<url>.*?)\1', webpage, 'formats', group='url', default=None)
-        if not video_url:
-            video_url = self._download_json(
-                'http://www.tvp.pl/pub/stat/videofileinfo?video_id=%s' % video_id,
-                video_id)['video_url']
-
-        ext = video_url.rsplit('.', 1)[-1]
-        if ext != 'ism/manifest':
-            if '/' in ext:
-                ext = 'mp4'
-            formats = [{
-                'format_id': 'direct',
-                'url': video_url,
-                'ext': ext,
-            }]
+        mime_ext = {
+            'video/x-ms-wmv': 'wmv',
+            'video/mp4': 'mp4'
+        }
+
+        viable_formats = []
+        for f in formats:
+            if f['mimeType'] in IGNORED_MIMETYPES:
+                continue
+
+            elif f['mimeType'].startswith('video/'):
+                viable_formats.append(
+                    {'url': f['url'],
+                     'ext': mime_ext.get(f['mimeType'], None),
+                     'vbr': f['totalBitrate']})
+
+        return viable_formats
+
+    def _get_video(self, context):
+        id = context['material_id']
+        if context['title_root']:
+            title = context['title_root']
+        elif not context['website_title']:
+            title = context['title']
         else:
-            m3u8_url = re.sub('([^/]*)\.ism/manifest', r'\1.ism/\1.m3u8', video_url)
-            formats = self._extract_m3u8_formats(m3u8_url, video_id, 'mp4')
+            title = ', '.join([context['website_title'], context['title']])
+        url = context['url']
+        description = context['description_root']
+
+        try:
+            formats = self._format_formats(self.api.formats(id), id)
+        except ExtractorError as e:
+            self.to_screen("%s: %s" % (title, e))
+            raise
 
         self._sort_formats(formats)
 
         return {
-            'id': video_id,
+            'id': str(id),
+            'url': url,
             'title': title,
-            'thumbnail': thumbnail,
+            'description': description,
             'formats': formats,
         }
 
+    def _get_playlist_videos(self, playlist_id):
+        ids = [playlist_id]
+
+        while ids:
+            item_id = ids.pop()
+            listing = self.api.listing(item_id)
+            for item in listing['items']:
+                if 'directory_video' in item['types']:
+                    ids.append(item['_id'])
+                if 'video' in item['types'] and item['is_released']:
+                    meta = self.api.context(item['_id'])
+                    try:
+                        yield self._get_video(meta)
+                    except ExtractorError:
```

Comment:
This defeats the whole point of `ExtractorError`. It should be propagated and handled outside the extractor not muted in extractor.